javascripts/adaptive-placeholder.js

```diff
@@ -0,0 +1,20 @@
+app.adaptivePlaceholder = {
+
+  changeFilledState: function () {
+    var input = $(this);
+    var value = input.val();
+    
+    if(value != '' && value != undefined) {
```

Comment:
I understand that, however, it doesn't check the types at all and for that reason can really get you into trouble, for example put these into a console:

```
0 == ''
0 === ''
false != ''
false !== ''
```
The number case could come back and bite you in this particular code section. Yes, it's a corner case, but certainly something we should be considering. Food for thought: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness and http://dorey.github.io/JavaScript-Equality-Table/

If you feel like we're good using `!=` here, okay, that's your call.